src/openrct2-ui/windows/Map.cpp

```diff
@@ -1189,89 +1189,92 @@ static void window_map_set_land_rights_tool_update(int32_t x, int32_t y)
  *
  *  rct2: 0x00666EEF
  */
-static void place_park_entrance_get_map_position(
-    int32_t x, int32_t y, int16_t* mapX, int16_t* mapY, int16_t* mapZ, int32_t* direction)
+static CoordsXYZD place_park_entrance_get_map_position(ScreenCoordsXY screenCoords)
 {
-    sub_68A15E(x, y, mapX, mapY);
-    if (*mapX == LOCATION_NULL)
-        return;
-
-    auto surfaceElement = map_get_surface_element_at(*mapX >> 5, *mapY >> 5);
-    *mapZ = surfaceElement->GetWaterHeight();
-    if (*mapZ == 0)
+    CoordsXYZD parkEntranceMapPosition{};
+    const CoordsXY mapCoords = sub_68A15E(screenCoords);
+    parkEntranceMapPosition = { mapCoords.x, mapCoords.y, parkEntranceMapPosition.z, parkEntranceMapPosition.direction };
+    if (parkEntranceMapPosition.x == LOCATION_NULL)
+        return parkEntranceMapPosition;
+
+    auto surfaceElement = map_get_surface_element_at(parkEntranceMapPosition.x >> 5, parkEntranceMapPosition.y >> 5);
+    parkEntranceMapPosition.z = surfaceElement->GetWaterHeight();
+    if (parkEntranceMapPosition.z == 0)
```

Comment:
All of this z code should be setting the value in terms of big coordinates not micro. Will need the appropriate *8's